controllers/workspace/devworkspace_controller.go

```diff
@@ -67,6 +70,219 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	// Validation limits for init-persistent-home container Command field
+	maxCommandElements      = 10
+	maxCommandElementLength = 1024
+
+	// Validation limits for init-persistent-home container Args field
+	maxArgsElements      = 50
+	maxArgsElementLength = 256 * 1024 // 256KB
+)
+
+// validateNoAdvancedFields validates that the init-persistent-home container
+// does not use advanced Kubernetes container fields that could make behavior unpredictable.
+func validateNoAdvancedFields(c corev1.Container) error {
```

Comment:
Can we move `validateImageReference` and move the  `validateNoAdvancedFields`,  `validateHomeInitContainer` logic to a different file instead of in `devworkspace_controller.go`?